### PR TITLE
Extensions: add ttmath support for fmod

### PR DESCRIPTION
**Pull request by mkaravel**
Add support for the `fmod()` free function.

### CODE REVIEW

**Comment by mkaravel** [comment]
As this is a modification on extensions, I think it makes sense to merge it after we are done with boost 1.58.

File: include/boost/geometry/extensions/contrib/ttmath_stub.hpp [hunk 1]
```diff
@@ -103,6 +107,12 @@ namespace ttmath
         return two * ATan((sqrt(x * x + y * y) - x) / y);
     }
 
+    // needed in order to work with boost::geometry::math::fmod
+    template <uint Exponent, uint Mantissa>
+    inline Big<Exponent, Mantissa> fmod(Big<Exponent, Mantissa> const& x, Big<Exponent, Mantissa> const& y)
```
Comment (awulkiew) [inline include/boost/geometry/extensions/contrib/ttmath_stub.hpp]: After the changes done in the other PR here the name should be changed to `mod()` as well.

Comment (mkaravel) [inline include/boost/geometry/extensions/contrib/ttmath_stub.hpp]: I am aware of that.
I was waiting to have the okay for the other PRs before modifying this.
I was also considering closing this one and creating a new one so as to keep the history simple.
Will re-act on it once the other PRs are okay/done.

**Comment by barendgehrels** [comment]
OK for me (with the name mod)

**Comment by mkaravel** [comment]
Function renamed from `fmod()` to `mod()`. See commit aac92cb.

**Pull request merged**